### ISSUE TITLE
Assist agent/customer to prevent duplicate ticket creation

### ISSUE THREAD

**Issue by dominikklein** [zammad/zammad#4592]
## Actual behaviour

Currently, when creating tickets, there is no check to see if a ticket has already been opened for this issue, device, contract, ... (or or or) has already been opened.

Such a check must be performed by the agent or customer, which entails the following risks:

a) An agent or customer can forget this check.
b) An agent or customer has to start an extended ticket search for such a check (time-consuming) and search for "attribute: value" or even "attribute1: value1 AND attribute2: value2" in case of multiple fields or more complex.
c) An agent or customer might not have access to corresponding tickets due to missing permissions.

In all cases, there is additional work in the organization, which could be prevented by a "simple" check during ticket creation (fully automatic) by Zammad.

## Expected behaviour:

**As an agent**
...I want to get notified by Zammad before creating a ticket about duplicate ticket creation

**As a customer**
...I want to get notified by Zammad before creating a ticket about duplicate ticket creation

**As an admin**
...I want to configure if this feature is enabled and what attributes should get used for the checks.

## Possible use cases

a) A "food delivery service" (or photo delivery, or web store, ...) issues vouchers for deliveries that were "faulty". For this purpose, the invoice number is stored in a ticket attribute (`invoice no`). If a customer reports to a delivery/invoice more than once (consciously or unconsciously), the customer can currently obtain an advantage, because the agent does not know whether there was already a support ticket for the order in question.

b) The customers of "IT Support" create tickets via Zammad. A certain classification is required when creating a ticket. If a ticket already exists for a certain classification, e.g. "Internet failure" or "Printer on 2nd floor does not work", the customer could be informed that a process already exists for this.

c) An "Internet Provider" classifies its tickets according to nodes that are stored at the ticket. If a ticket has already been created for a node, the problem is already known.

d) A "hardware manufacturer" stores a serial number of the affected devices on the ticket. If there is already a service request in the form of a ticket, the agent knows immediately that it is a double message.

e) A "fire department" stores the vehicle and the service concerned on tickets. If another shift should e.g. report the failure of the company radio in a certain vehicle, then multiple creations of processes can also be made here. 

## Possible Solution

On the ticket create screen there could be an automatic lookup before the ticket is created if a ticket with the same attributes already exists.

### Agent & Customer Interface

The conditions should be checked on the backend side to find all relevant tickets (with the configured authorization lookup).

Proposal: The check should only happen on field changes for the configured "field to be checked".

### Config Options

- Ticket attributes to be checked
- Title of the message
- Text of the message
- Limit showing of the message to certain user roles only
- If the tickets in question are to be listed in the popup
- Lookup authorization (user authorization or system authorization)
- Should be searched in open or also in closed tickets

# Mockups

![Zammad-Test-System-Ticket](https://user-images.githubusercontent.com/6049445/236146930-79e9cb36-4a48-4a3b-a6ae-3d9c1ee122bc.png)

![Screen Shot 2023-04-26 at 12 30 21](https://user-images.githubusercontent.com/6049445/234562484-fca398c8-fe74-44be-bdc1-b624997b56af.png)

![Screen Shot 2023-04-26 at 12 30 36](https://user-images.githubusercontent.com/6049445/234562521-f0202505-0cec-4ae6-9498-034d2039dc22.png)

![Screen Shot 2023-04-26 at 12 31 40](https://user-images.githubusercontent.com/6049445/234562562-28c0331e-0796-4c97-baf8-1b18c3e0ccbb.png)

<img width="385" src="https://user-images.githubusercontent.com/6049445/234562613-36de1054-499e-4bcb-a111-bedf4e20ad4e.png">

```[tasklist]
### Tasks
- [x] desktop | Add new UI element to show the information in the form instead of the popup. (@dvuckovic)
- [x] desktop | Move controller to core workflow to save requests. (@rolfschmidt)
- [x] desktop | Consider to support for boolean/date/datetime/integer. (@rolfschmidt)
- [x] desktop | add setting to search for open/closed tickets. (@rolfschmidt)
- [x] desktop | Update mockups with the new approach and reword texts. (@dvuckovic)
- [x] desktop | Rename all settings/variables to not have the wording popup. (@dvuckovic)
- [x] desktop | Drop old code like controllers or request tests. (@dvuckovic)
- [x] desktop | Add UI element tests. (@dvuckovic)
- [x] desktop | Add core workflow tests. (@rolfschmidt)
- [x] desktop | Add capybara tests. (@dvuckovic)
- [x] desktop | Rename feature, because it is not preventing anything. (@dvuckovic)
- [x] mobile | Add new UI element to show the information in a popup. (@dvuckovic)
- [x] mobile | Add core workflow support for the UI element. (@dvuckovic)
- [x] mobile | Add frontend tests. (@dvuckovic)
- [x] feat | Fix permission issue with leaking ticket information (@dvuckovic)
- [x] feat | Add a setting to limit feature to specific user roles (@dvuckovic)
- [x] Review
```

**Comment by tschaefer** [comment]
**Implementation acceptance criteria**

* In the first iteration, we do duplication checks on simple single-input ticket attributes only.

**Comment by rolfschmidt** [comment]
> **Implementation acceptance criteria**
> 
> * In the first iteration, we do duplication checks on simple single-input ticket attributes only.

supported are now `'input', 'select', 'boolean', 'date', 'datetime', 'integer'`. multi tree select and multiselect are blocked by a naming issue (https://github.com/zammad/coordination-technical-debt/issues/290) and postponed for now.

**Comment by Bozo0511** [comment]
- I tested every Attribute to compare
- I tested the duplicate prevention for all tickets and then only for open tickets.
- After that, I checked the duplicate Ticket prevention for different roles 

I could not find any issues, the entire Feature is working as intended
